### PR TITLE
handle internal names and JvmName annotations for name mangling.

### PR DESCRIPTION
**Pull request by neetopia**
putting it in draft mode since this PR only fixes source symbols. Theoretically it should handle library symbols if psi stub creation for library symbols are in place (as per the document of the psi elements in analysis API symbols). The fix of this PR can be seen by enabling `mangledNames` test, JvmName annotations and internal names should be handled now.

Note that light class still won't handle inline names.

### CODE REVIEW

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/ResolverAAImpl.kt [hunk 1]
```diff
@@ -480,8 +483,18 @@ class ResolverAAImpl(
...
 
-    // TODO: handle @JvmName annotations, mangled names
+    // TODO: handle library symbols
     override fun getJvmName(accessor: KSPropertyAccessor): String? {
+        (
+            (accessor.receiver.closestClassDeclaration() as? KSClassDeclarationImpl)
+                ?.ktClassOrObjectSymbol?.psi as? KtClassOrObject
+            )?.toLightClass()?.allMethods?.filterIsInstance<SymbolLightSimpleMethod>()
+            ?.singleOrNull {
+                (it.parameters.isNotEmpty() xor (accessor is KSPropertyGetter)) &&
```
Comment (neetopia) [inline kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/ResolverAAImpl.kt:493]: This line is very tricky, the idea is we don't have a way to tell if a light method is getter or setter, and both getter and setter shares a same property declaration as the `kotlinOrigin`, so we just check if an accessor has any parameters to determine whether it is a getter or setter.

**Comment by ting-yuan** [comment]
> Theoretically it should handle library symbols if psi stub creation for library symbols are in place (as per the document of the psi elements in analysis API symbols)

Will this ever be available in standalone mode? Does it require indexing of binary dependencies?

**Comment by neetopia** [comment]
> > Theoretically it should handle library symbols if psi stub creation for library symbols are in place (as per the document of the psi elements in analysis API symbols)
> 
> Will this ever be available in standalone mode? Does it require indexing of binary dependencies?



> > Theoretically it should handle library symbols if psi stub creation for library symbols are in place (as per the document of the psi elements in analysis API symbols)
> 
> Will this ever be available in standalone mode? Does it require indexing of binary dependencies?

That's something we need to investigate further. Currently there are already library symbols with a stub in KSP2, see https://github.com/google/ksp/pull/1873/

**Review by ting-yuan — APPROVED** [review]
Feel free to merge this first and address the remaining cases later.

**Pull request merged**